misc/python/materialize/cloudtest/k8s/environmentd.py

```diff
@@ -148,6 +148,8 @@ def generate_stateful_set(self) -> V1StatefulSet:
             "--adapter-stash-url=postgres://root@cockroach.default:26257?options=--search_path=adapter",
             "--storage-stash-url=postgres://root@cockroach.default:26257?options=--search_path=storage",
             "--internal-sql-listen-addr=0.0.0.0:6877",
+            "--internal-persist-pubsub-listen-addr=0.0.0.0:6879",
```

Comment:
```
[pod/environmentd-0/environmentd] error: Found argument '--internal-persist-pubsub-listen-addr' which wasn't expected, or isn't valid in this context
[pod/environmentd-0/environmentd] 
[pod/environmentd-0/environmentd] 	Did you mean '--internal-sql-listen-addr'?
```

Reply:
was this locally or in CI?

Reply:
In CI: https://buildkite.com/materialize/nightlies/builds/2349#01880fb6-4bf8-49f9-a751-ee68f9bb328a
See [kubectl-get-logs.log](https://buildkite.com/organizations/materialize/pipelines/nightlies/builds/2349/jobs/01880fb6-4bf8-49f9-a751-ee68f9bb328a/artifacts/01880fc0-618e-41a1-a980-bc016aa40852)

Reply:
I am so puzzled by this one, it works fine locally for me and the error output there doesn't even list the other `--persist-pubsub-url` param that's added in

```
environmentd --orchestrator <ORCHESTRATOR> --persist-blob-url <PERSIST_BLOB_URL> --persist-consensus-url <PERSIST_CONSENSUS_URL> --storage-stash-url <POSTGRES_URL> --adapter-stash-url <POSTGRES_URL> --environment-id <<CLOUD>-<REGION>-<ORG-ID>-<ORDINAL>> --sql-listen-addr <HOST:PORT> --http-listen-addr <HOST:PORT> --availability-zone <AVAILABILITY_ZONE> --aws-account-id <AWS_ACCOUNT_ID> --aws-external-id-prefix <ID> --announce-egress-ip <ANNOUNCE_EGRESS_IP>... --orchestrator-kubernetes-image-pull-policy <ORCHESTRATOR_KUBERNETES_IMAGE_PULL_POLICY> --internal-sql-listen-addr <HOST:PORT>
```

Reply:
that one is part of the `main` entrypoint to environmentd https://github.com/MaterializeInc/materialize/pull/18981/files#diff-18db540d7592188856fb9af92ebb418ad154a7b8c04e88c798f33a99b530c0a6R221-R232

Reply:
internal-sql-listen-addr comes from `src/environmentd/src/bin/environmentd/main.rs`: 
```
    #[clap(
        long,
        value_name = "HOST:PORT",
        env = "INTERNAL_SQL_LISTEN_ADDR",
        default_value = "127.0.0.1:6877"
    )]
    internal_sql_listen_addr: SocketAddr,
```

Reply:
Cloudtest is passing for the current build in CI: https://buildkite.com/materialize/tests/builds/55421#01880fc9-cedf-4efc-a6c0-1d81fc2d246a

Reply:
Pretty puzzled by that error, I didn't see anything odd in your commits but it looks like that envd was running without both new params

Reply:
Same nightly "Platform upgrade checks for Cloudtest/k8s" is failing on this build, but if you check the `kubectl describe all` logs the `environmentd` image running on the pods isn't the same as the one from the build... I think we might have a bug in the test harness

Reply:
OH it's because the test is trying to run an upgrade compatibility check with the last release, but is using the new version's args. @def- let me know if you have suggestions here!

Reply:
👍 trying that now in nightlies 🤞

Reply:
that fixed it! @def-, paul had to take off for the day so i'm taking this over, but still ramping up on the context. what else would we like to do before getting this merged?

Reply:
Great, my final step will be the same test with the flag on: https://buildkite.com/materialize/nightlies/builds/2357
If this is green I'm out of ideas for now, so good from my side.

Reply:
The upgrade test is green now.

Reply:
the other test you kicked off in that run failed, but looks like it's the same errors we're seeing in main